src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -565,7 +565,7 @@ def get(self, args):
             return None
 
         # Get the top-level command, like `network` in `network vnet create -h`
-        top_command = args[0]
+        top_command = args[0].lower()
```

Comment:
As you commented in https://github.com/Azure/azure-cli/issues/27497#issuecomment-1752373586, knack also has the logic to convert command to low case:

https://github.com/microsoft/knack/blob/e0c14114aea5e4416c70a77623e403773aba73a8/knack/invocation.py#L87

I haven't debugged into the source code, but why isn't `args` here already converted to lower case?

Reply:
Knack's convert happens in `CLI.invoke()`.
In this line, args is still case-sensitive.

Reply:
`args` is used to call `load_command_table` on each module:

https://github.com/Azure/azure-cli/blob/15296fd2604cb98cd8bd4a56d7a6d3c65bd291bd/src/azure-cli-core/azure/cli/core/commands/__init__.py#L1067

such as in https://github.com/Azure/azure-cli/pull/27571#discussion_r1360207782.

If `args` is not standardized in a centralized location, its case (lower or upper) may cause other problem besides command index.